src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const states = [];
+
+  for (const action of actions) {
+    let newState = { ...currentState };
```

Comment:
According to the checklist, use a more descriptive name for the object copy. Instead of `newState`, a name like `stateCopy` would better convey the purpose of this variable.